arduino/libraries/librariesmanager/download.go

```diff
@@ -17,16 +17,13 @@ package librariesmanager
 
 import (
 	"net/url"
-
-	"go.bug.st/downloader/v2"
 )
 
-// LibraryIndexURL is the URL where to get library index.
+// LibraryIndexURL is the URL where to get the library index.
 var LibraryIndexURL, _ = url.Parse("https://downloads.arduino.cc/libraries/library_index.json")
 
-// UpdateIndex downloads the libraries index file from Arduino repository.
-func (lm *LibrariesManager) UpdateIndex(config *downloader.Config) (*downloader.Downloader, error) {
```

Comment:
Is this considered a breaking change?

Reply:
In theory, yes, but it was a so weird function that I doubt anyone has ever used it alone.
I'll add a note in the UPGRADING.md

Reply:
Haha, yeah I took a long dive down that rabbit hole when I was switching to use Arduino CLI's library index infrastructure in Arduino Lint (https://github.com/arduino/arduino-lint/commit/455001dfa1234abad1137389a22ab89c298c7cce), I ended up deciding to stick to the pre-existing bespoke code approach for the downloading after I got hopelessly lost, so Arduino Lint is not affected by this change.

I will take another look at what Arduino CLI has to offer if I get an opportunity to switch Arduino Lint to downloading the gzipped index.